### ISSUE TITLE
unnested_or_patterns trigger on master build even if feature is not enabled

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#5704]
When using the master toolchain and a clippy built by master, the `unnested_or_patterns` lint triggers on crates although the `or_patterns` feature is not enabled in these crates.

This does not happen if the nightly-clippy is used.

See https://github.com/rust-lang/rust-clippy/pull/5378#issuecomment-641208834 for details

**Comment by CAD97** [comment]
I just saw this lint with `cargo +nightly-2020-06-16 clippy`, so I think this is triggering in nightly-clippy as well now.

**Comment by repi** [comment]
Ran into this with nightly (`clippy 0.0.212 (e55d3f9c5 2020-06-18)`) now also, this clippy suggestions:

```rust
warning: unnested or-patterns
   --> components\module-host\src\host_api\world\components\component_accessors.rs:701:13
    |
701 | /             (D6Joint::RelativePosition, _)
702 | |             | (D6Joint::RelativeRotation, _)
703 | |             | (D6Joint::RelativeAngles, _) => {
    | |__________________________________________^
    |
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unnested_or_patterns
help: nest the patterns
    |
701 |             (D6Joint::RelativePosition | D6Joint::RelativeRotation |
702 |  D6Joint::RelativeAngles, _) => {
```

But changing the code to this does indeed fail as it is an experimental feature we do not have enabled:

```rust
error[E0658]: or-patterns syntax is experimental
   --> components\module-host\src\host_api\world\components\component_accessors.rs:702:17
    |
702 |                 D6Joint::RelativePosition | D6Joint::RelativeRotation | D6Joint::RelativeAngles,
    |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: see issue #54883 <https://github.com/rust-lang/rust/issues/54883> for more information
    = help: add `#![feature(or_patterns)]` to the crate attributes to enable
```
